sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -22,18 +22,87 @@ import java.util.Properties
+ *      path gets reported for the input_file_name properly without having to convert
+ *      paths back to the original. This also has the benefit that it can be more performant
+ *      if it doesn't have to do the extra list leaf files, but you don't get the
+ *      locality information updated. So for small Alluxio clusters or with Spark
+ *      clusters short on task slots this may be a better fit.
+ *
+ * The way we do the actual replacement algorithm differs depending on the file reader
+ * type we use: PERFILE, COALESCING or MULTITHREADED.
+ * PERFILE is not supported with Alluxio due to not easily being able to fix up
+ * input_file_name. We could but would require copying the FileScanRDD so skip for now.
+ * The COALESCING reader is not support when input_file_name is requested so it falls
+ * back to the MULTITHREADED reader if that is used, when input_file_name is not requested,
+ * we replace the paths properly based on the replacement algorithm and don't have to worry
+ * about calculating the original path.  The MULTITHREADED reader supports input_file_name
+ * so it handles calculating the original file path in the case of the conver time algorithm.
+ * In order to do the replacement at task time and to output the original path for convert
+ * time, we need to have a mapping of the original scheme to the alluxio scheme. This has been
+ * made a parameter to many of the readers. With auto mount and task time replacement,
+ * we make a pass through the files on the driver side in GpuFileSourceScanExec in order to
+ * do the mounting before the tasks try to access alluxio.
+ * Note that Delta Lake uses the input_file_name functionality to do things like
+ * Updates and Deletes and will fail if the path has the alluxio:// in it.
+ *
+ * Below we support 2 configs to turn on Alluxio, we have the automount which uses a regex
+ * to replace paths and then we have the config that specifies direct paths to replace and
+ * user has to manually mount those.
+ */
 object AlluxioUtils extends Logging {
   private val checkedAlluxioPath = scala.collection.mutable.HashSet[String]()
+  private val ALLUXIO_SCHEME = "alluxio://"
+  private val mountedBuckets: scala.collection.mutable.Map[String, String] =
+    scala.collection.mutable.Map()
+  private var alluxioCmd: Seq[String] = null
+  private var alluxioMasterHost: Option[String] = None
+  private var alluxioPathsToReplaceMap: Option[Map[String, String]] = None
+  private var alluxioHome: String = "/opt/alluxio-2.8.0"
+  private var isInit: Boolean = false
+
+  def checkAlluxioNotSupported(rapidsConf: RapidsConf): Unit = {
+    if (rapidsConf.isParquetPerFileReadEnabled &&
+      (rapidsConf.getAlluxioAutoMountEnabled || rapidsConf.getAlluxioPathsToReplace.isDefined)) {
+      throw new IllegalArgumentException("Alluxio is currently not supported with the PERFILE " +
+        "reader, please use one of the other reader types.")
```

Comment:
would be nice if this said what reader to use? i.e. (MULTITHREADED or COALESCING)

Reply:
should we silently fallback to MULTITHREADED or COALESCING by default? and warn about it ... I don't know what the right answer is.

Reply:
I'd rather just error because PERFILE isn't the default so user likely set it for specific reason like working around another problem.